src/buildercore/command.py

```diff
@@ -0,0 +1,125 @@
+# Fabric 1.14 documentation: https://docs.fabfile.org/en/1.14/
+
+import sys
+import fabric.api as fab_api
+import fabric.contrib.files as fab_files
+import fabric.exceptions as fab_exceptions
+import fabric.state
+import logging
+from io import BytesIO
+from . import utils
+
+LOG = logging.getLogger(__name__)
+
+env = fab_api.env
+
+#
+# exceptions
+#
+
+class CommandException(Exception):
+    pass
+
+# lsh@2019-10: deprecated in favour of CommandException
+class FabricException(CommandException):
+    pass
+
+# no un-catchable errors from Fabric
+env.abort_exception = FabricException
+
+NetworkError = fab_exceptions.NetworkError
+
+#
+# api
+#
+
+local = fab_api.local
+execute = fab_api.execute
+parallel = fab_api.parallel
+serial = fab_api.serial
+hide = fab_api.hide
+
+# https://github.com/mathiasertl/fabric/blob/master/fabric/context_managers.py#L158-L241
+def settings(*args, **kwargs):
+    "a context manager that alters mutable application state for functions called within it's scope"
+
+    # these values were set with `fabric.state.output[key] = val`
+    # they would be persistant until the program exited
+    # - https://github.com/mathiasertl/fabric/blob/master/fabric/state.py#L448-L474
+    for key, val in kwargs.pop('fabric.state.output', {}).items():
+        fabric.state.output[key] = val
+
+    return fab_api.settings(*args, **kwargs)
+
+lcd = fab_api.lcd # local change dir
+rcd = fab_api.cd # remote change dir
+
+remote = fab_api.run
+remote_sudo = fab_api.sudo
+upload = fab_api.put
+download = fab_api.get
+remote_file_exists = fab_files.exists
+
+# https://github.com/mathiasertl/fabric/blob/master/fabric/utils.py#L30-L63
+def abort(msg):
+    msg = "\nFatal error: %s\nAborting" % msg
+    sys.stderr.write(msg)
+    sys.stderr.flush()
+    exit(1)
```

Comment:
shouldn't this use `buildercore.config.env.FabricException` which has been deleted?

Reply:
FabricException hasn't been deleted, just moved to `command.py`

Our direct usage of `abort` (in `cfn.py`) is so simple I just recreated it. The `abort` implementation referenced above the function is slightly more complex and will raise the configured `abort_exception` (`FabricException`). *That* `abort` and that behaviour isn't being. `command.py` just gathers our Fabric usage into one place.